ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -81,6 +82,12 @@ public StreamedQueryResource(
 
   @POST
   public Response streamQuery(final KsqlRequest request) throws Exception {
+    if (!ksqlEngine.isAcceptingStatements()) {
```

Comment:
I think we should add a similar check to `WSQueryEndpoint`? Doesn't seem like we should accept new query requests via websockets either if we're in the process of terminating the cluster.